core/ledger/pvtdatastorage/store.go

```diff
@@ -1218,6 +1222,16 @@ func (p *purgeUpdatesProcessor) process(hashedIndexKey, hashedIndexVal []byte) e
 			break
 		}
 	}
+
+	// If configured, log the purged key for audit purpose
+	if p.purgedKeyAuditLogging {
+		decodedDataKey, err := decodeDatakey(dataKey)
+		if err != nil {
+			return err
+		}
+		logger.Infow("Processed private data purge from private data storage", "channel", ledgerid, "chaincode", decodedDataKey.nsCollBlk.ns, "collection", decodedDataKey.nsCollBlk.coll, "key", string(hashedIndexVal), "blockNum", decodedDataKey.nsCollBlk.blkNum, "tranNum", decodedDataKey.txNum)
+	}
```

Comment:
As this is being highlighted a audit log, this requires some discussion to be aware of some scenario where it may report falsely.

The log at this place would be a premature. Here, the data for deletion is getting prepared in-memory (imagine a crash scenario). The alternate place for this log would be in the function after 
```
if err := p.db.WriteBatch(p.batch, true); err != nil {
		return err
}
``` 
in the funciton `commitPendingChanges` below. Even a better place could be at the end of function `deleteDataMarkedForPurge`.

In order to be able to do this in the above mentioned function, we would need to keep track of the keys getting deleted -- we can introduce a map in this struct for this purpose. Of course, this could create a reverse problem of not printing even after deleting -- I am not sure if we would want to care about that situation, but that can be handled by leaving this current log here and changing it's tone to "going to purge..."

Further, we can limit the number of logs per key (e.g., not to repeat the log for a key for each block but instead only for the highest block as user just wants to know what keys were purged in a cycle)

Also, about printing the private key in the logs, we can instead just print the hash of the key (in hex) so we don't create another problem for folks to purge from logs.

Reply:
Makes sense... to keep the code most simple, I went with the approach of logging "Purging private data..." for each key as it is processed (if audit logging is enabled). Then conclude with "Purged private data..." with summary counts after the purges are all committed. This way the user sees the keys if they want to, and gets a final confirmation that they were are actually purged. This will also help us in support scenarios.

In terms of logging the actual key versus a hash, my logging philosophy has always been that we can safely log keys, but not values. If we want to scrub keys from log files, I think that would be a much wider change across many peer log statements. Users have two options with the current approach - they can exclude actual private data from the key, or they can disable audit logging. Let's see if that is sufficient.